Rubberduck.VBEditor.VBA/SafeComWrappers/VB/VBE.cs

```diff
@@ -207,7 +207,7 @@ private static bool ComponentHasName(IVBComponent c, string name)
             {"SLDWORKS.EXE", typeof(SolidWorksApp)},
         };
 
-        private static IHostApplication _host;
+        private static IHostApplication _host = null;
 
         /// <summary> Returns the type of Office Application that is hosting the VBE. </summary>
         public IHostApplication HostApplication()
```

Comment:
This is technically outside the scope of the PR, but I think we need to recognize that this method is doing wayyy too much and should be extracted out into at least 2 private methods (e.g. `DiscoverHostApplicationViaCommands` and `DiscoverHostApplicationViaReferences`), assign to the `_host` once. I am not sure if we also should be retrying the discovery as long the `_host` is `null` since it looks like if the result is `null`, the `_host` is left `null`, and next call will start the discovery process anew. Maybe that is necessary, in which case we need to make it more obvious. (e.g. do a `bool TryGetHostApplication(out IHostApplication hostApplication)` instead of `IHostApplication HostApplication`)